src/test/java/seedu/address/model/person/DeliveryDateTest.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import org.junit.jupiter.api.Test;
+
+
+
+public class DeliveryDateTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeliveryDate(null));
+    }
+
+    @Test
+    public void constructor_invalidDeliveryDate_throwsIllegalArgumentException() {
+        String invalidDeliveryDate = "2021/01";
+        assertThrows(IllegalArgumentException.class, () -> new DeliveryDate(invalidDeliveryDate));
+    }
+
+    @Test
+    public void isValidDeliveryDate() {
+        // null delivery date
+        assertThrows(NullPointerException.class, () -> DeliveryDate.isValidDeliveryDate(null));
+
+        // invalid addresses
+        assertFalse(DeliveryDate.isValidDeliveryDate("01/13/2022")); // invalid month
+        assertFalse(DeliveryDate.isValidDeliveryDate("32/01/2022")); // invalid day
+        assertFalse(DeliveryDate.isValidDeliveryDate("01/01/202020")); // invalid year
+        // less than 3 working days
+        assertFalse(DeliveryDate.isValidDeliveryDate(LocalDate.now().plusDays(2L).toString()));
```

Comment:
Are there any delimiters your delivery date class is intentionally not meant to handle? If so perhaps you can add an invalid delimiter test here

Reply:
Yup, there are. Nice catch again:)